pkg/onboarding/presentation/rest/handlers_helpers.go

```diff
@@ -51,3 +54,25 @@ func decodeOTPPayload(
 
 	return payload, nil
 }
+
+func respondWithError(w http.ResponseWriter, code int, err error) {
+	errMap := errorcodeutil.ErrorMap(err)
+	errBytes, err := json.Marshal(errMap)
+	if err != nil {
+		errBytes = []byte(fmt.Sprintf("error: %s", err))
+	}
+	respondWithJSON(w, code, errBytes)
+}
+
+func respondWithJSON(w http.ResponseWriter, code int, payload []byte) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	_, err := w.Write(payload)
+	if err != nil {
+		log.Printf(
+			"unable to write payload `%s` to the http.ResponseWriter: %s",
+			string(payload),
+			err,
+		)
+	}
+}
```

Comment:
DRY this code. These functions have been implemented in one of our libraries. Ask @NYARAS which library they are in.